src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -874,6 +879,81 @@ export class StudyViewPageStore {
         });
     }
 
+    private createCnaGeneComparisonSession(
+        chartMeta: ChartMeta,
+        hugoGeneSymbols: string[],
+        statusCallback: (phase: LoadingPhase) => void
+    ) {
+        statusCallback(LoadingPhase.DOWNLOADING_GROUPS);
+
+        // Get mutations among currently selected samples
+        const promises: any[] = [this.selectedSamples, this.cnaProfiles];
+
+        return new Promise<string>(resolve => {
+            onMobxPromise<any>(
+                promises,
+                async (
+                    selectedSamples: Sample[],
+                    cnaProfiles: MolecularProfile[]
+                ) => {
+                    const studyToCnaProfile = _.keyBy(
+                        cnaProfiles,
+                        p => p.studyId
+                    );
+                    const sampleMolecularIdentifiers = selectedSamples.reduce(
+                        (array, sample) => {
+                            if (sample.studyId in studyToCnaProfile) {
+                                array.push({
+                                    sampleId: sample.sampleId,
+                                    molecularProfileId:
+                                        studyToCnaProfile[sample.studyId]
+                                            .molecularProfileId,
+                                });
+                            }
+                            return array;
+                        },
+                        [] as SampleMolecularIdentifier[]
+                    );
+
+                    const genes = await client.fetchGenesUsingPOST({
+                        geneIdType: 'HUGO_GENE_SYMBOL',
+                        geneIds: hugoGeneSymbols,
+                    });
+
+                    const cnaByGeneAndAlteration = _.groupBy(
+                        (
+                            await client.fetchMolecularDataInMultipleMolecularProfilesUsingPOST(
```

Comment:
kinda don't like the practice of embedding awaits inside of logic like this.  it just obscures the async nature.   i know asyn/await is supposed to smooth async coding patterns but i still think it's nice to assign it to a variable and then deal with that variable.   more explicit.